runtime/runtime-params-estimator/src/main.rs

```diff
@@ -102,49 +107,48 @@ fn main() -> anyhow::Result<()> {
         Some(it) => it,
         None => {
             temp_dir = tempfile::tempdir()?;
-
-            let contract_code = read_resource(if cfg!(feature = "nightly_protocol_features") {
-                "test-contract/res/nightly_small_contract.wasm"
-            } else {
-                "test-contract/res/stable_small_contract.wasm"
-            });
-
-            let state_dump_path = temp_dir.path().to_path_buf();
-            nearcore::init_configs(
-                &state_dump_path,
-                None,
-                Some("test.near".parse().unwrap()),
-                Some("alice.near"),
-                1,
-                true,
-                None,
-                false,
-                None,
-                false,
-                None,
-                None,
-                None,
-            )
-            .expect("failed to init config");
-
-            let near_config = load_config(&state_dump_path, GenesisValidationMode::Full);
-            let store = create_store(&get_store_path(&state_dump_path));
-            GenesisBuilder::from_config_and_store(
-                &state_dump_path,
-                Arc::new(near_config.genesis),
-                store,
-            )
-            .add_additional_accounts(cli_args.additional_accounts_num)
-            .add_additional_accounts_contract(contract_code)
-            .print_progress()
-            .build()
-            .unwrap()
-            .dump_state()
-            .unwrap();
-
-            state_dump_path
+            temp_dir.path().to_path_buf()
         }
     };
+    if init_state {
+        let contract_code = read_resource(if cfg!(feature = "nightly_protocol_features") {
+            "test-contract/res/nightly_small_contract.wasm"
+        } else {
+            "test-contract/res/stable_small_contract.wasm"
+        });
+
+        nearcore::init_configs(
+            &state_dump_path,
+            None,
+            Some("test.near".parse().unwrap()),
+            Some("alice.near"),
+            1,
+            true,
+            None,
+            false,
+            None,
+            false,
+            None,
+            None,
+            None,
+        )
+        .expect("failed to init config");
+
+        let near_config = load_config(&state_dump_path, GenesisValidationMode::Full);
+        let store = create_store(&get_store_path(&state_dump_path));
+        GenesisBuilder::from_config_and_store(
+            &state_dump_path,
+            Arc::new(near_config.genesis),
+            store,
+        )
+        .add_additional_accounts(cli_args.additional_accounts_num)
+        .add_additional_accounts_contract(contract_code)
+        .print_progress()
+        .build()
+        .unwrap()
+        .dump_state()
+        .unwrap();
```

Comment:
I wonder if we can `?` now

Reply:
Not out of the box:
```
error[E0277]: the size for values of type `dyn std::error::Error` cannot be known at compilation time
   --> runtime/runtime-params-estimator/src/main.rs:149:22
    |
149 |         .dump_state()?;
    |                      ^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `dyn std::error::Error`
    = note: required because of the requirements on the impl of `std::error::Error` for `Box<dyn std::error::Error>`
    = note: required because of the requirements on the impl of `From<Box<dyn std::error::Error>>` for `anyhow::Error`
    = note: required because of the requirements on the impl of `FromResidual<Result<Infallible, Box<dyn std::error::Error>>>` for `Result<(), anyhow::Error>`

error[E0277]: `dyn std::error::Error` cannot be sent between threads safely
   --> runtime/runtime-params-estimator/src/main.rs:149:22
    |
149 |         .dump_state()?;
    |                      ^ `dyn std::error::Error` cannot be sent between threads safely
    |
    = help: the trait `Send` is not implemented for `dyn std::error::Error`
    = note: required because of the requirements on the impl of `Send` for `Unique<dyn std::error::Error>`
    = note: required because it appears within the type `Box<dyn std::error::Error>`
    = note: required because of the requirements on the impl of `From<Box<dyn std::error::Error>>` for `anyhow::Error`
    = note: required because of the requirements on the impl of `FromResidual<Result<Infallible, Box<dyn std::error::Error>>>` for `Result<(), anyhow::Error>`
```

Reply:
Ah yeah, that's the general problem that, due to coherence, there may only be *one* dynamically typed error. ideally we should migrate that other crate to anyhow.